pennylane_lightning/src/StateVector.hpp

```diff
@@ -140,6 +140,39 @@ template <class fp_t = double> class StateVector {
         gate(internalIndices, externalIndices, inverse, params);
     }
 
+    /**
+     * @brief Apply a single gate to the state-vector.
+     *
+     * @param matrix Arbitrary unitary gate to apply.
+     * @param wires Wires to apply gate to.
+     * @param inverse Indicates whether to use inverse of gate.
+     * @param params Optional parameter list for parametric gates.
+     */
+    void applyOperation(const std::vector<CFP_t> &matrix,
+                        const vector<size_t> &wires, bool inverse = false,
+                        [[maybe_unused]] const vector<fp_t> &params = {}) {
```

Comment:
Yea, the params here is to allow simplified dispatch functionality. We define `std::function` with a given signature elsewhere, and attempt to ensure similarity throughout. Keeping the params in there was just a cheap way of ensuring this with no additional overheads, since `[[maybe_unused]]` is now fully supported by the standard.